common/helpers/g2p.py

```diff
@@ -0,0 +1,101 @@
+__all__ = ["G2PBasedOovAugmenter"]
+
+import os
+from typing import Optional
+
+from sisyphus import tk
+Path = tk.setup_path(__package__)
+
+from i6_core.corpus.stats import ExtractOovWordsFromCorpusJob
+from i6_core.g2p.apply import ApplyG2PModelJob
+from i6_core.g2p.convert import BlissLexiconToG2PLexiconJob, G2POutputToBlissLexiconJob
+from i6_core.g2p.train import TrainG2PModelJob
+
+
+class G2PBasedOovAugmenter():
+    def __init__(self,
+                 original_bliss_lexicon:str,
+                 train_lexicon:Optional[str] = None,
+                 g2p_model_path:Optional[Path] = None,
+                 train_args:Optional[dict] = None,
+                 apply_args:Optional[dict] = None,
+                 ):
+        """
+        :param original_bliss_lexicon: path to the original lexicon that will be augmented with OOVs
+        :param train_lexicon: path to the train lexicon in case it differs from the original lexicon
+        :param g2p_model_path: path to the g2p model, if None a g2p model is trained
+        #######################################
+        :param train_args = {
+        "num_ramp_ups"   :4,
+        "min_iter"       :1,
+        "max_iter"       :60,
+        "devel"          :"5%",
+        "size_constrains":"0,1,0,1"
+        }
+        #######################################
+        :param apply_args = {
+        "variants_mass"  :1.0,
+        "variants_number":1
+        }
+        """
+        super().__init__()
+        self.original_bliss_lexicon = original_bliss_lexicon
+        self.train_lexicon = original_bliss_lexicon if train_lexicon is None else train_lexicon
+        self.g2p_model_path = g2p_model_path
+        self.train_args = {}
+        self.apply_args = {}
+
+        if train_args is not None:
+            self.train_args.update(train_args)
+
+        if apply_args is not None:
+            self.apply_args.update(apply_args)
```

Comment:
this was much shorter before and did the same thing, or did you want to add some defaults? If the dicts are empty anyway you do not need to call update, but just keep the one-line if/else statement.

Reply:
This has not been changed from the first commit of the class. I don't understand what you mean. Here, the jobs have their own default parameters. However, one might change a value. The default arguments in both cases are `None`, so I cannot initialize the `self.{train,apply}_args` with the passed parameters, however I would need to update the dict if the user passed something.

> keep the one-line if/else statement.
What do you mean online if/else statement?

Reply:
Oh sorry, I thought it was different before. I mean:
`self.train_args = train_args if train_args else {}`

Reply:
I can also do it like this, if you prefer.

Reply:
I am just saying it can be shorter, but you can decide

Reply:
ok. done